src/header/CatalogHeader.test.tsx

```diff
@@ -0,0 +1,170 @@
+import { getConfig } from '@edx/frontend-platform';
+
+import { render } from '../setupTest';
+import { ROUTES } from '../routes';
+import CatalogHeader from './CatalogHeader';
+import { useMenuItems } from './hooks/useMenuItems';
+import messages from './messages';
+
+jest.mock('@edx/frontend-platform', () => ({
+  getConfig: jest.fn(() => ({
+    LMS_BASE_URL: process.env.LMS_BASE_URL,
+    ENABLE_PROGRAMS: process.env.ENABLE_PROGRAMS,
+    ENABLE_COURSE_DISCOVERY: process.env.ENABLE_COURSE_DISCOVERY,
+    SUPPORT_URL: process.env.SUPPORT_URL,
+  })),
+  mergeConfig: jest.fn(),
+  ensureConfig: jest.fn(),
+  subscribe: jest.fn(),
+}));
+
+jest.mock('./hooks/useMenuItems', () => ({
+  useMenuItems: jest.fn(),
+}));
+
+describe('CatalogHeader', () => {
+  const mockMenuItems = {
+    mainMenu: [
+      {
+        type: 'item',
+        href: `${getConfig().LMS_BASE_URL}/dashboard`,
+        content: messages.courses.defaultMessage,
+      },
+    ],
+    secondaryMenu: [
+      {
+        type: 'item',
+        href: getConfig().SUPPORT_URL,
+        content: messages.help.defaultMessage,
+      },
+    ],
+  };
+
+  beforeEach(() => {
+    (useMenuItems as jest.Mock).mockReturnValue(mockMenuItems);
+  });
+
+  afterEach(() => jest.clearAllMocks());
+
+  it('renders header component', () => {
+    render(<CatalogHeader />);
+
+    expect(document.body).toBeInTheDocument();
+  });
+
+  it('calls useMenuItems hook', () => {
+    render(<CatalogHeader />);
+
+    expect(useMenuItems).toHaveBeenCalled();
+  });
+
+  it('passes correct props to Header component', () => {
+    const { mainMenu, secondaryMenu } = mockMenuItems;
+
+    render(<CatalogHeader />);
+
+    expect(useMenuItems).toHaveBeenCalled();
+    const hookResult = (useMenuItems as jest.Mock).mock.results[0].value;
+    expect(hookResult.mainMenu).toEqual(mainMenu);
+    expect(hookResult.secondaryMenu).toEqual(secondaryMenu);
+  });
+
+  it('handles different menu configurations', () => {
+    const mockMenuItemsWithPrograms = {
+      mainMenu: [
+        ...mockMenuItems.mainMenu,
+        {
+          type: 'item',
+          href: `${getConfig().LMS_BASE_URL}/programs`,
+          content: messages.programs.defaultMessage,
+        },
+      ],
+      secondaryMenu: mockMenuItems.secondaryMenu,
+    };
+
+    (useMenuItems as jest.Mock).mockReturnValue(mockMenuItemsWithPrograms);
+
+    render(<CatalogHeader />);
+
+    const hookResult = (useMenuItems as jest.Mock).mock.results[0].value;
+    expect(hookResult.mainMenu).toHaveLength(2);
+    expect(hookResult.mainMenu[1].content).toBe(messages.programs.defaultMessage);
```

Comment:
It feels odd to me to rely on watching the hooks instead of seeing what is rendered. I pulled this down locally and put `screen.debug()` before and after the `render(<CatalogHeader />)` line and got

Before:
```html
<body />
```

After:
```html
<body>
  <div />
</body>
```

which also seems odd.

So it seems these tests aren't verifying the full story of "when `useMenuItems` returns data as we expect it to, it is rendered properly in the header."

I'm not sure what the best path forward for this is. It seems [for learner dash the tests are just mocking `frontend-component-header`](https://github.com/openedx/frontend-app-learner-dashboard/blob/master/src/containers/LearnerDashboardHeader/index.test.jsx), so maybe going back to that makes more sense here?

Reply:
> I added [some changes](https://github.com/openedx/frontend-app-catalog/pull/24/commits/190d8e761256c6a051329815eea2a58c4adcc25d), similar to the Learner Dashboard MFE

The changes look good to me! While I'd still rather be able to have the tests verify that the actual header components renders everything as expected, matching the pattern used in `frontend-app-learner-dashboard` seems like a great solution for now.

> I noticed that the [frontend-component-header](https://github.com/openedx/frontend-component-header) repository doesn't have as many unit tests as we would like. Maybe it's worth creating a separate issue for this.

That sounds like a good plan to me!